contrib/win32/win32compat/win32_usertoken_utils.c

```diff
@@ -811,10 +811,18 @@ windows_password_auth(const char *username, const char* password)
 		}
 	}
 
-	if (pLogonUserExExW(unam_utf16, udom_utf16, pwd_utf16, LOGON32_LOGON_NETWORK_CLEARTEXT,
-		LOGON32_PROVIDER_DEFAULT, NULL, &token, NULL, NULL, NULL, NULL) == TRUE)
+	if (pLogonUserExExW(unam_utf16, udom_utf16, pwd_utf16, LOGON32_LOGON_INTERACTIVE,
+		LOGON32_PROVIDER_WINNT50, NULL, &token, NULL, NULL, NULL, NULL) == TRUE) {
 		password_auth_token = token;
-	else {
+		debug("Password-based logon (interactive) succeeded for user: %ls domain: %ls",
+			unam_utf16, udom_utf16);
+	} else if (GetLastError() ==  ERROR_LOGON_TYPE_NOT_GRANTED && 
+		pLogonUserExExW(unam_utf16, udom_utf16, pwd_utf16, LOGON32_LOGON_NETWORK_CLEARTEXT,
+		LOGON32_PROVIDER_WINNT50, NULL, &token, NULL, NULL, NULL, NULL) == TRUE) {
```

Comment:
Does it have to be LOGON32_PROVIDER_DEFAULT?

Reply:
According to the documentation, using LOGON32_PROVIDER_DEFAULT uses NTLM which can be less insecure.  Using LOGON32_PROVIDER_WINNT50 prefers Kerberos which is preferred from a security perspective.  Some companies have even disabled NTLM on their systems.

Reply:
Is that true, according to https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-logonuserw `LOGON32_PROVIDER_DEFAULT` is Negotiate


> Use the standard logon provider for the system. The default security provider is negotiate, unless you pass NULL for the domain name and the user name is not in UPN format. In this case, the default provider is NTLM.

The only caveat to that is if the domain is `NULL` and the username is not in the UPN format.

Reply:
@jborean93 I'm not sure if LogonUserExEx ends up calling LogonUser under the convers, but it does not have that same behavior noted according to it's documentation https://docs.microsoft.com/en-us/windows/win32/secauthn/logonuserexexw:
> Use the standard logon provider for the system. The default security provider is NTLM.
>
Notably, I did not trace it enable/analyze the security logs so I suspect it's you're very possibly right, but probably doesn't hurt to be explicit.